subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -748,6 +748,38 @@ static int tls_opt_ciphersuite_list_get(struct net_context *context,
 	return 0;
 }
 
+static int tls_opt_ciphersuite_used_get(struct net_context *context,
+					void *optval, socklen_t *optlen)
+{
+	const char *ciph;
+
+	if (!context || !context->tls) {
+		return -EBADF;
+	}
+
+	if (!optval || !optlen) {
+		return -EFAULT;
+	}
+
+	if (*optlen == 0) {
+		return 0;
```

Comment:
Is this right return code?

Reply:
I'd say -EINVAL whenever the option length is not what is expected by a socket function.